tsl/src/planner.c

```diff
@@ -179,6 +180,34 @@ copy_mode_enabled(void)
 	return strcmp(enable_copy, "true") == 0;
 }
 
+/* Query tree walker to examine RTE contents for a distributed hypertable */
+static bool
+distributed_rtes_walker(Node *node, bool *isdistributed)
+{
+	/* if isdistributed is already set, then no need to walk further */
+	if (*isdistributed)
+		return true;
+	if (node == NULL)
+		return false;
+	if (IsA(node, RangeTblEntry))
+	{
+		RangeTblEntry *rte = (RangeTblEntry *) node;
+
+		if (rte->rtekind == RTE_RELATION)
+			ts_rte_is_hypertable(rte, isdistributed);
+		return false;
```

Comment:
Nit: you can abort the walk here immediately if `*isdistributed` is set to true. It is not touched anywhere else. That would also mean that you do not need the if in the beginning of the function.
```suggestion
		if (rte->rtekind == RTE_RELATION)
			ts_rte_is_hypertable(rte, isdistributed);
		return *isdistributed;
```

Reply:
Done.